src/sfizz/HistoricalBuffer.h

```diff
@@ -60,11 +62,18 @@ class HistoricalBuffer {
      */
 	ValueType getAverage() const
 	{
-		return mean<ValueType>(buffer);
+        if (!validMean) {
+            mean = sfz::mean<ValueType>(buffer);
+            validMean = true;
+        }
```

Comment:
You can track the power in more efficient ways which don't involve O(n) computation.

- one is always to keep the moving sum so you skip the O(n) completely.
  As a value gets pushed, remove from the moving sum the buffer's oldest value, and add the new one.
  A problem is that this can drift. A solution is to use fixed-point integers instead of floats, which make it exact.
  Then you can easily make the length way >16, which currently isn't really much.

- otherwise to track the signal level, another possibility commonly used:
  take the absolute values of input samples, pass it in lowpass filter that has different response times on rising and falling signal. It should be immediate on rising and slow on falling.
similar to this https://fr.wikipedia.org/wiki/Redresseur#Filtrage_de_la_tension_de_sortie_d'un_pont_de_diode

Reply:
Ok my bad I misunderstood on first read.
Regardless the current solution faces a tiny problem, it's the power computation will be different depending how you feed signal into the library, where in small buffers or large ones.
So the solutions above are relevant in this regard.